modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -404,6 +404,21 @@ public ObjectDefinition putObjectDefinition(
 					serviceBuilderObjectField.getBusinessType(),
 					ObjectFieldConstants.BUSINESS_TYPE_RELATIONSHIP)) {
 
+				String dbTableName = serviceBuilderObjectField.getDBTableName();
+
+				if (serviceBuilderObjectDefinition.isApproved()) {
+					dbTableName =
+						serviceBuilderObjectDefinition.
+							getExtensionDBTableName();
+				}
+				else if (Validator.isNull(dbTableName)) {
+					dbTableName =
+						serviceBuilderObjectDefinition.getDBTableName();
+				}
+
+				_objectFieldLocalService.updateDBTableName(
+					dbTableName, serviceBuilderObjectField.getObjectFieldId());
```

Comment:
@seltong as mentioned in the previous comment, the ideal would be to move this logic to the `localService`